src/utils/date.ts

```diff
@@ -10,4 +37,125 @@ export function isDate(dateString?: ValueType) {
   }
 
   return !isNaN(new Date(dateString).getDate());
-}
\ No newline at end of file
+}
+
+export enum DATE_FORMATS {
+  TIME = "h:mm A",
+  SHORT = "MMM DD",
+  LONG = "h:mm A, MMM DD, YYYY",
+  TIME_RANGE = "YYYY-MM-DD HH:mm"
+}
+
+type formatDateOptions = {
+  suffix?: string;
+  asTimestamp?: boolean;
+  stringFormat?: DATE_FORMATS;
+};
+
+/**
+ * Format date into local time and apply the string formatting if given
+ * else the date object is parsed into local time
+ *
+ * @param date date which needs to be parsed or formatted
+ * @param stringFormat string of tokens to format the date, check https://day.js.org/docs/en/display/format#docsNav
+ * @returns string if stringFormat is passed else a date object
+ */
+export const formatDate = (
+  date: string | Date,
+  { asTimestamp, stringFormat, suffix }: formatDateOptions = {}
+) => {
+  const localDate = dayjs(date).local();
+  if (asTimestamp) return localDate.valueOf().toString();
+  const formattedDate = stringFormat
+    ? localDate.format(stringFormat)
+    : localDate;
+  return suffix ? `${formattedDate}${suffix}` : formattedDate;
+};
+
+/**
+ * Parse the given date relative to the user locale
+ *
+ * @param date
+ * @param date
+ * @returns relative date as string
+ */
+export const relativeDateTime = (
+  from: string | Date,
+  to = dayjs().toISOString()
+) => {
+  const fromDate = dayjs.utc(from).local();
+  const toDate = dayjs.utc(to).local();
+
+  if (
+    toDate.isSame(from, "day") &&
+    toDate.isSame(from, "month") &&
+    toDate.isSame(from, "year")
+  ) {
+    return fromDate.fromNow();
+  }
+
+  if (
+    toDate.diff(fromDate, "days") === 1 &&
+    toDate.isSame(from, "month") &&
+    toDate.isSame(from, "year")
+  ) {
+    return `${fromDate.format(DATE_FORMATS.TIME)} yesterday`;
+  }
+
+  return fromDate.format(DATE_FORMATS.LONG);
+};
+
+/**
+ *
+ * @param value date to check
+ * @returns boolean check if value is a valid date
+ */
+export const isValidDate = (value: DateConfig) => dayjs(value).isValid();
+
+/**
+ *
+ * @param value date to convert
+ * @returns string of ISO date
+ */
+export const dateToISOString = (value: DateConfig) =>
+  dayjs(value).toISOString();
+
+/**
+ *
+ * @param value date to convert
+ * @returns Native date object
+ */
+export const dateToJSDate = (value: DateConfig) => dayjs(value).toDate();
+
+/**
+ *
+ * @param value date to format
+ * @param format custom string format for the date
+ * @returns formatted string of the date
+ */
+export const customDateFormattor = (value: DateConfig, format: string) =>
```

Comment:
Why do we need this ?

Reply:
We have created TimeRangePicker component which similar to Grafana time range picker this function is being used in that component & Canary Status Chart xAxis value formatting.

Reply:
Then it should in a formatFunction, we don't want to distribute date formats throughout the code base

Reply:
Done